tests/app/views/test_services.py

```diff
@@ -490,6 +493,191 @@ def test_should_400_if_mismatched_service_id(self):
                   response.get_data())
 
 
+class TestShouldCallSearchApiOnPutToCreateService(BaseApplicationTest):
+    def setup(self):
+        super(TestShouldCallSearchApiOnPutToCreateService, self).setup()
+        with self.app.app_context():
+            db.session.add(
+                Framework(id=1, expired=False, name="G-Cloud 6")
+            )
+            db.session.add(
+                Supplier(supplier_id=1, name=u"Supplier 1")
+            )
+
+            db.session.commit()
+
+    def test_should_index_on_service_put(self):
+        with self.app.app_context():
+            search_api_client.index = Mock(return_value=True)
+
+            payload = self.load_example_listing("G6-IaaS")
+            payload['id'] = "1234567890123456"
+            self.client.put(
+                '/services/1234567890123456',
+                data=json.dumps(
+                    {
+                        'update_details': {
+                            'updated_by': 'joeblogs',
+                            'update_reason': 'whateves'},
+                        'services': payload}
+                ),
+                content_type='application/json')
+
+            service = Service.query.filter(Service.service_id ==
+                                           "1234567890123456").first()
+            search_api_client.index.assert_called_with(
+                "1234567890123456",
+                service.data,
+                "Supplier 1"
+            )
+
+
+class TestShouldCallSearchApiOnPutToReplaceService(BaseApplicationTest):
+    def setup(self):
+        super(TestShouldCallSearchApiOnPutToReplaceService, self).setup()
+        now = datetime.now()
+        payload = self.load_example_listing("G6-IaaS")
+        with self.app.app_context():
+            db.session.add(
+                Framework(id=1, expired=False, name="G-Cloud 6")
+            )
+            db.session.add(
+                Supplier(supplier_id=1, name=u"Supplier 1")
+            )
```

Comment:
Seeing as how my pull request has recently been merged, it's worth pointing out that Suppliers will not be returned from the database if they haven't got any Contact Information ([as it's modelled using a Inner Join](https://github.com/alphagov/digitalmarketplace-api/blob/master/app/models.py#L118)).  So all of the tests that expect a Supplier to be returned will fail.  

I added Contact Information to [unit tests that create Suppliers](https://github.com/alphagov/digitalmarketplace-api/blob/master/tests/app/views/test_services.py#L170), but these newer ones don't have them.  

(I can add them in for you if you like.)